backend/question/routes/questionRoutes.ts

```diff
@@ -0,0 +1,137 @@
+import express from 'express';
+import { check, validationResult, body } from 'express-validator';
+import Question from "../models/Question";
+
+/**
+ * Router for the question service.
+ */
+
+const router = express.Router();
+
+router.get('/', (req, res) => {
+    res.send('Hello from question service!');
+});
+
+// Create a new question
+router.post('/create', [
+    check('title').isString().isLength({ min: 1 }),
+    check('description').isString().isLength({ min: 1 }),
+    check('category').isString().isLength({ min: 1 }),
+    check('complexity').isString().isLength({ min: 1 }),
+], async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        return res.status(400).json({ errors: errors.array() });
+    }
+
+    try {
+        const { title, description, category, complexity } = req.body;
+        const question = { title, description, category, complexity };
+        const newQuestion = new Question(question);
+        await newQuestion.save();
+        res.status(200).json({ message: 'Question created succesfully', question: newQuestion });
+    } catch (error) {
+        res.status(500).send('Internal server error');
+    }
+});
+
+// Retrieve a specific question by id
+router.get('/:id', [
+    check('id').isNumeric(),
+], async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        return res.status(400).json({ errors: errors.array() });
+    }
+
+    const questionId = parseInt(req.params.id, 10);
+    try {
+        const question = await Question
+            .findOne({ questionid: questionId })
+            .exec();
+        if (question) {
+            res.json(question);
```

Comment:
Think you can use return statements for early termination so e.g return res.json(question) then don't need to use the else block